Sources/PackageCollections/Providers/GitHubPackageMetadataProvider.swift

```diff
@@ -0,0 +1,227 @@
+/*
+ This source file is part of the Swift.org open source project
+
+ Copyright (c) 2020 Apple Inc. and the Swift project authors
+ Licensed under Apache License v2.0 with Runtime Library Exception
+
+ See http://swift.org/LICENSE.txt for license information
+ See http://swift.org/CONTRIBUTORS.txt for Swift project authors
+ */
+
+import Basics
+import Dispatch
+import struct Foundation.Date
+import class Foundation.JSONDecoder
+import struct Foundation.NSRange
+import class Foundation.NSRegularExpression
+import struct Foundation.URL
+import PackageModel
+import TSCBasic
+
+struct GitHubPackageMetadataProvider: PackageMetadataProvider {
+    let configuration: Configuration
+    let httpClient: HTTPClient
+    let decoder: JSONDecoder
+    let queue: DispatchQueue
+
+    init(configuration: Configuration = .init(), httpClient: HTTPClient = .init()) {
+        self.configuration = configuration
+        self.httpClient = httpClient
+        self.decoder = JSONDecoder()
+        #if os(Linux)
```

Comment:
Do we need this extra case here? Linux should already end up in the first case of the other if statement, right?

Reply:
we actually do, since availability check does not cover linux